packages/prosemirror-resizable-view/src/prosemirror-resizable-view.ts

```diff
@@ -1,172 +1,278 @@
   // size when `update` method is called.
-  width = '';
+  #width = '';
+  #height = '';
 
   constructor({
     node,
     view,
     getPos,
-    createElement,
+    aspectRatio = ResizableRatioType.Fixed,
+    options,
+    initialSize,
   }: {
     node: ProsemirrorNode;
     view: EditorView;
     getPos: () => number;
-    createElement: CreateElement;
+    aspectRatio?: ResizableRatioType;
+    options?: OptionShape;
+    initialSize?: { width: number; height: number };
   }) {
-    const outer = document.createElement('div');
-    outer.style.position = 'relative';
-    outer.style.width = node.attrs.width;
-    outer.style.maxWidth = '100%';
-    outer.style.minWidth = '50px';
-    outer.style.display = 'inline-block';
-    outer.style.lineHeight = '0'; // necessary so the bottom right handle is aligned nicely
-    outer.style.transition = 'width 0.15s ease-out, height 0.15s ease-out'; // make sure transition time is larger then mousemove event's throttle time
+    const outer = this.createWrapper(node, initialSize);
+    const element = this.createElement({ node, view, getPos, options });
 
-    const inner = createElement({ node, view, getPos });
-
-    const handle = document.createElement('div');
-    handle.style.position = 'absolute';
-    handle.style.bottom = '0px';
-    handle.style.right = '0px';
-    handle.style.width = '16px';
-    handle.style.height = '16px';
-    handle.style.borderBottom = handle.style.borderRight = '4px solid #000';
-    handle.style.display = 'none';
-    handle.style.zIndex = '100';
-    handle.style.cursor = 'nwse-resize';
-
-    const setHandleDisplay = (visible?: true | string) => {
-      visible = visible || handle.dataset.mouseover || handle.dataset.dragging;
-      handle.style.display = visible ? '' : 'none';
-    };
+    const handleRight = new ResizableHandle(ResizableHandleType.Right);
+    const handleLeft = new ResizableHandle(ResizableHandleType.Left);
 
     outer.addEventListener('mouseover', () => {
-      handle.dataset.mouseover = 'true';
-      setHandleDisplay(true);
+      handleRight.setHandleDisplay(true);
+      handleLeft.setHandleDisplay(true);
     });
 
     outer.addEventListener('mouseout', () => {
-      handle.dataset.mouseover = '';
-      setHandleDisplay();
+      handleRight.setHandleDisplay();
+      handleLeft.setHandleDisplay();
     });
 
-    handle.addEventListener('mousedown', (e: MouseEvent) => {
-      e.preventDefault();
+    handleRight.dom.addEventListener('mousedown', (e) =>
+      this.resizeHandler(e, view, getPos, handleRight),
+    );
+    handleLeft.dom.addEventListener('mousedown', (e) =>
+      this.resizeHandler(e, view, getPos, handleLeft),
+    );
+
+    outer.append(handleRight.dom);
+    outer.append(handleLeft.dom);
+
+    /**
+     * Only initialize `handleBottom`, `handleBottomRight`, and `handleBottomLeft` in
+     * `ResizableRatioType.Flexible` mode
+     */
```

Comment:
I'm thinking if we should give users more flexibility to choose handles. For example, what if a user only wants the two corner handles (bottomRight and bottomLeft)?

Reply:
Because in different ResizableRatioType the horizontal handles have different behavior.  So I think we could better just offer several general options, and the arguments could be neat. 
And we could extend any general type in the future, if needed.